lib/JsonClient.js

```diff
@@ -98,6 +105,14 @@ JsonClient.prototype.parse = function parse(req, callback) {
             }
         }
 
+        // if no http error but we had a json parse error, return the json
+        // parse err as the top level error
+        if (!resErr && parseErr) {
+            resErr = new RestError(parseErr, {
+                message: 'Invalid JSON in response'
```

Comment:
I propose we should try and send the actual raw malformed string that we were unable to parse. It is helpful for debugging.

Reply:
The original error (`parseErr`) is used as a cause (first parameter) for this wrapped error - you'll still get the full message and stack trace from the original error.